pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -568,6 +570,8 @@ func (s *streamIngestionResumer) OnFailOrCancel(
 	// ingestion anymore.
 	jobExecCtx := execCtx.(sql.JobExecContext)
 	completeProducerJob(ctx, s.job, jobExecCtx.ExecCfg().InternalDB, false)
+	// On a job fail or cancel, replication has permanently stopped so set replicated time to 0
```

Comment:
nit: mention that this could occasionally get overriden because of the race with processor shut down.

Reply:
Done